sql/log_event.h

```diff
@@ -2706,11 +2807,15 @@ class Unknown_log_event: public binary_log::Unknown_event , public Log_event
   Unknown_log_event(const char* buf,
                     const Format_description_event *description_event)
   : binary_log::Unknown_event(buf, description_event),
-    Log_event(header(), footer())
+    Log_event(header(), footer()), what(UNKNOWN)
   {
     is_valid_param= true;
   }
 
+  /* constructor for hopelessly corrupted events */
+  Unknown_log_event() : Log_event(header(), footer()), what(ENCRYPTED)
```

Comment:
We are not sure, but this is how they decided to solve the problem of mysql client tools bumping into encrypted event. This Uknown_log_event is only created by client tools when force option is used. Then it shows those events as encrypted, although there is a chance they are not encrypted. However it will be hard for those tools to be able to distinguished between hopelessly corrupted event and encrypted event  - so it seems like a good approximation.